src/number_theory/arithmetic_function.lean

```diff
@@ -231,5 +242,269 @@ instance [comm_ring α] : comm_ring (arithmetic_function α) :=
 { .. arithmetic_function.add_comm_group,
   .. arithmetic_function.comm_semiring }
 
+section pointwise_mul
+
+/-- This is the pointwise product of `arithmetic_function`s. -/
+def pointwise_mul [mul_zero_class α] (f g : arithmetic_function α) :
+  arithmetic_function α :=
+⟨λ x, f x * g x, by simp⟩
+
+@[simp]
+lemma pointwise_mul_apply [mul_zero_class α] {f g : arithmetic_function α} {x : ℕ} :
+  (pointwise_mul f g) x = f x * g x := rfl
+
+lemma pointwise_mul_comm [comm_monoid_with_zero α] (f g : arithmetic_function α) :
+  pointwise_mul f g = pointwise_mul g f :=
+by { ext, simp [mul_comm] }
+
+variable [monoid_with_zero α]
+
+/-- This is the pointwise power of `arithmetic_function`s. -/
+def pointwise_pow (f : arithmetic_function α) (k : ℕ) (kpos : 0 < k) :
+  arithmetic_function α :=
+⟨λ x, (f x) ^ k, by { rw [map_zero], exact zero_pow kpos }⟩
+
+@[simp]
+lemma pointwise_pow_apply {f : arithmetic_function α} {k x : ℕ} {kpos : 0 < k} :
+  (pointwise_pow f k kpos) x = (f x) ^ k := rfl
+
+lemma pointwise_pow_succ {f : arithmetic_function α} {k : ℕ} {kpos : 0 < k} :
+  pointwise_pow f (k + 1) (k.succ_pos) = pointwise_mul f (pointwise_pow f k kpos) :=
+by { ext, simp [pow_succ] }
+
+lemma pointwise_pow_succ' {f : arithmetic_function α} {k : ℕ} {kpos : 0 < k} :
+  pointwise_pow f (k + 1) (k.succ_pos) = pointwise_mul (pointwise_pow f k kpos) f :=
+by { ext, simp [pow_succ'] }
+
+end pointwise_mul
+
+section multiplicative
+
+section monoid_with_zero
+variable [monoid_with_zero α]
+
+/-- Multiplicative functions -/
+class is_multiplicative  (f : arithmetic_function α) : Prop :=
```

Comment:
We have tried such classes with group and ring homomorphisms, and they don't play well with `simp`.
If you make this a structure, and I have `h : f.is_multiplicative`, then I can write `simp [h.map_one]` and it will work.
But with your current approach, I think it will turn out to be a gamble whether simp can find the instance that it needs.

Reply:
Would you recommend bundling this, or making it a regular prop?

Reply:
I would go for a regular prop, like with continuous functions.